File: 04-ORIENTACAO_A_OBJETOS/JavaScript - NodeJS/Exercicio 5/Com Pacotes/Veiculo.js
module.exports = class Veiculo {
    constructor(marca, cor, motorLigado, estilo){
        this.marca = marca
        this.cor = cor
        this.motorLigado = motorLigado
        this.estilo = estilo
    }

    ligaDesligaMotor(acao){
        if (acao)
            console.log('Motor está ligado')
        else
            console.log('Motor está desligado')
    }
    
    mostraAtributos(){
        return this.marca+' '+this.cor+' '+this.estilo
    }
}

File: 04-ORIENTACAO_A_OBJETOS/Java/Exercicio 5/Veiculo.java
/*

 */
public class Veiculo {
    String marca;
    String cor;
    boolean motorLigado;
    
    Veiculo(String marca, String cor, boolean motorLigado){
        this.marca = marca;
        this.cor = cor;
        this.motorLigado = motorLigado;
    }
    
    boolean isMotorLigado(){
      return this.motorLigado;
    }

    void ligaMotor(){
        if(this.motorLigado == true)
            System.out.println("O motor está ligado");
        else{
            this.motorLigado = true;
            System.out.println("O motor acaba de ser ligado");
        }
    }
    
}
